sdk/core/System.ClientModel/src/Pipeline/ClientPipeline.cs

```diff
@@ -64,6 +64,8 @@ public static ClientPipeline Create(ClientPipelineOptions options, params Pipeli
     {
         Argument.AssertNotNull(options, nameof(options));
 
+        options.Freeze();
```

Comment:
If we decide all ClientModel-based clients will freeze options, I think it's ok to freeze it in the pipeline.Create method rather than requiring the client to do it.  For Azure.Core-based clients, we can let the client decide whether to Freeze the options or not based on whether this would be a breaking change.